### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by CharelMarais**
Pr to merge TS changes and convert the project to TS

### CODE REVIEW

File: src/dom-manipulation.ts [hunk 1]
```diff
@@ -19,109 +20,112 @@ export function appendFlightInformationToFlightInfoContainer(flight) {
...
-        flight[0] + (flight[1] ?? 0)
+        flight.icao24 + (flight.callsign ?? 0)
       }" class="track-button uppercase font-bold cursor-pointer border-none">track</button> 
     </div>`;
   }
 }
 
-function appendExistingFlightInfoRow(flight) {
-  const exsitingFlightInfoRow = document.getElementById(flight[0]);
+function appendExistingFlightInfoRow(flight: IFlights) {
```
Comment (MihlaliMgwebi) [inline src/dom-manipulation.ts]: strongly type your functions that don't return a value e.g. void.

File: src/dom-manipulation.ts [hunk 2]
```diff
@@ -19,109 +20,112 @@ export function appendFlightInformationToFlightInfoContainer(flight) {
...
     // if a button is clicked
-    mapElement.style.visibility = "visible";
-    button.innerText = "close";
-    setMapAndMarkerToCurrentFlightLocation(fltInfo[6], fltInfo[5], fltInfo[10]);
+    mapElement!.style.visibility = "visible";
+    button!.innerText = "close";
+    setMapAndMarkerToCurrentFlightLocation(fltInfo.latitude, fltInfo.longitude, fltInfo.true_track);
   }
-  showAndHideButtonsAfterClick(button.innerText);
+  showAndHideButtonsAfterClick(button!.innerText);
```
Comment (MihlaliMgwebi) [inline src/dom-manipulation.ts:128]: Be careful of the bang operator. Use null checks instead if possible

**Review by MihlaliMgwebi — COMMENTED** [review]
Looks good (love the usage of type assertion)

Just check my comments.

File: src/dom-manipulation.ts [hunk 3]
```diff
@@ -7,9 +8,9 @@ import {
   convertMeterPerSecondToKilomentersPerHour,
 } from "./utils/utils.js";
 
-export function appendFlightInformationToFlightInfoContainer(flight) {
+export function appendFlightInformationToFlightInfoContainer(flight: IFlights) {
```
Comment (kphutiyagae) [inline src/dom-manipulation.ts]: Possible spelling error on IFlights?

**Review by kphutiyagae — CHANGES_REQUESTED** [review]

File: src/dom-manipulation.ts (hunk 1 above)
Comment (CharelMarais) [reply src/dom-manipulation.ts]: I have added :void to strongly type my non returning functions

File: src/dom-manipulation.ts (hunk 2 above)
Comment (CharelMarais) [reply src/dom-manipulation.ts:128]: I reworked the button bang operator. The bang I used below this is for an event that will exist without fail same with the parent element.

File: src/dom-manipulation.ts (hunk 3 above)
Comment (CharelMarais) [reply src/dom-manipulation.ts]: Looks good to me, where do you see the mistake so I can resolve it.

**Pull request merged**